artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -1249,6 +1256,14 @@ public RoutingStatus send(final ServerMessage message, final boolean direct) thr
 
    @Override
    public RoutingStatus send(final ServerMessage message, final boolean direct, boolean noAutoCreateQueue) throws Exception {
+
+      // If the protocol doesn't support flow control, we have no choice other than fail the communication
+      if (!this.getRemotingConnection().isSupportsFlowControl() && pagingManager.isDiskFull()) {
+         ActiveMQIOErrorException exception = ActiveMQMessageBundle.BUNDLE.diskBeyondLimit();
+         this.getRemotingConnection().fail(exception);
+         throw exception;
```

Comment:
Why are these the only 2 solutions?  Surely we want to keep consumers alive so we can gracefully recover from the situation?  

Killing the connections could kill all the consumers too making it difficult to recover once the memory and disk limit is reached.  DROP doesn't necessarily mean the messages are lost, the client should choose an appropriate qos level and retry later.  Is it possible to configure it to go from PAGING to DROP once the disk limit is reached?